src/ldk.rs

```diff
@@ -697,7 +741,138 @@ async fn handle_ldk_events(
+                    .channel_manager
+                    .fail_intercepted_htlc(intercept_id)
+                    .unwrap();
+            }
+
+            let get_rgb_info = |channel_id| {
+                let is_colored =
+                    is_channel_rgb(channel_id, &PathBuf::from(&static_state.ldk_data_dir));
+                if is_colored {
+                    let (rgb_info, _) = get_rgb_channel_info(
+                        channel_id,
+                        &PathBuf::from(&static_state.ldk_data_dir),
+                    );
+                    Some((
+                        rgb_info.contract_id,
+                        rgb_info.local_rgb_amount,
+                        rgb_info.remote_rgb_amount,
+                    ))
+                } else {
+                    None
+                }
+            };
+
+            let inbound_channel = unlocked_state
+                .channel_manager
+                .list_channels()
+                .into_iter()
+                .find(|details| details.short_channel_id == Some(prev_short_channel_id))
+                .expect("Should always be a valid channel");
+            let outbound_channel = unlocked_state
+                .channel_manager
+                .list_channels()
+                .into_iter()
+                .find(|details| details.short_channel_id == Some(requested_next_hop_scid))
+                .expect("Should always be a valid channel");
+
+            let inbound_rgb_info = get_rgb_info(&inbound_channel.channel_id);
+            let outbound_rgb_info = get_rgb_info(&outbound_channel.channel_id);
+
+            tracing::debug!("EVENT: Requested swap with params inbound_msat={} outbound_msat={} inbound_rgb={:?} outbound_rgb={:?} inbound_contract_id={:?}, outbound_contract_id={:?}", inbound_amount_msat, expected_outbound_amount_msat, inbound_rgb_amount, expected_outbound_rgb_amount, inbound_rgb_info.map(|i| i.0), outbound_rgb_info.map(|i| i.0));
+
+            let mut trades_lock = unlocked_state.taker_trades.lock().unwrap();
+            let (whitelist_contract_id, whitelist_swap_type) = match trades_lock.get(&payment_hash)
+            {
+                None => {
+                    tracing::error!("ERROR: rejecting non-whitelisted swap");
+                    unlocked_state
+                        .channel_manager
+                        .fail_intercepted_htlc(intercept_id)
+                        .unwrap();
+                    return;
+                }
+                Some(x) => x,
+            };
+
+            match whitelist_swap_type {
+                crate::swap::SwapType::BuyAsset {
+                    amount_rgb,
+                    amount_msats,
+                } => {
+                    // We subtract HTLC_MIN_MSAT because a node receiving an RGB payment also receives that amount of sats with it as the payment amount,
+                    // so we exclude it from the calculation of how many sats we are effectively giving out.
```

Comment:
How would you subtract the number of sats from the paid amount? The amount being paid is in some arbitrary asset and the node is not aware of any exchange rate for that asset vs sats. Moreover, it would require both nodes to exactly agree on that exchange rate, which could be complicated if you factor in latency, different price feeds, etc

Reply:
As discussed in another channel, it should be possible to avoid this substraction because the taker can always detect the amount sent for making the RGB payment possible (usually the `HTLC_MIN_MSAT`, but could be a higher value) and then send the amount of bitcoins specified in the swap string, considering this "underlying amount" part of the bitcoins that should be sent. The consequence of this approach is that the API that constructs the swap string should make sure that we're always swapping at least `HTLC_MIN_MSAT`.